shell/platform/android/io/flutter/embedding/engine/systemchannels/AccessibilityChannel.java

```diff
@@ -103,12 +103,12 @@ public void setAccessibilityMessageHandler(@Nullable AccessibilityMessageHandler
     void announce(@NonNull String message);
 
     /**
-     * The user has tapped on the artifact with the given {@code nodeId}.
+     * The user has tapped on the widget with the given {@code nodeId}.
```

Comment:
A widget may include a gesture detector, which creates a render object. That object lends its callbacks to a semantics node which the node id corresponds to.

There is no strong relationship between a widget and a semantics node. A widget may create multiple semantics nodes, or may only contribute some configuration to an existing node.

Reply:
That's a good point.  I'm wondering, from a developers perspective reading this, would this be a case where technically lying about the widget would still convey the right message?  We could replace "widget" with "renderobject", right?  But I'm wondering if that would be even more confusing?

Reply:
Technically it is another tree created by the render objects. The most correct description would be "semantics node" with perhaps some clarification that these aren't 1-1 with widgets or render objects

Reply:
A semantics node can also wrap many widgets, right?

Reply:
No, semantics nodes have no concept of widgets - their children are other semantics node. Not to be confused with the semantics widget :)